src/ui/pages/AssetInfo/AssetInfo.tsx

```diff
@@ -0,0 +1,253 @@
+import { useQuery } from '@tanstack/react-query';
+import { walletPort } from 'src/ui/shared/channels';
+import { useAddressParams } from 'src/ui/shared/user-address/useAddressParams';
+import { isReadonlyAccount } from 'src/shared/types/validators';
+import { useWalletAssetDetails } from 'src/modules/zerion-api/hooks/useWalletAssetDetails';
+import { useBackgroundKind } from 'src/ui/components/Background';
+import { UnstyledLink } from 'src/ui/ui-kit/UnstyledLink';
+import { useWalletPortfolio } from 'src/modules/zerion-api/hooks/useWalletPortfolio';
+import { useHttpClientSource } from 'src/modules/zerion-api/hooks/useHttpClientSource';
+import { NetworkId } from 'src/modules/networks/NetworkId';
+import { CircleSpinner } from 'src/ui/ui-kit/CircleSpinner';
+import * as styles from './styles.module.css';
+import { AssetHistory } from './AssetHistory';
+import { AssetAddressStats } from './AssetAddressDetails';
+import { AssetGlobalStats } from './AssetGlobalStats';
+import { AssetTitleAndChart } from './AssetTitleAndChart';
+import { AssetResources } from './AssetResources';
+import { AssetHeader } from './AssetHeader';
+import { AssetDescription } from './AssetDescription';
+
+const SCROLL_THRESHOLD = 80;
+const SHOW_BRIDGE_BUTTON = false; // TODO: make true after bridge is implemented
+
+function AssetPageHeader({ asset }: { asset: Asset }) {
+  const [showTokenInfoInHeader, setShowTokenInfoInHeader] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () =>
+      setShowTokenInfoInHeader(window.scrollY < SCROLL_THRESHOLD);
+
+    handleScroll();
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  return showTokenInfoInHeader ? null : (
+    <AssetHeader asset={asset} className={styles.assetHeaderContent} />
+  );
+}
+
+function ReportAssetLink({ asset }: { asset: Asset }) {
+  return (
+    <UnstyledAnchor
+      target="_blank"
+      href={`https://zerion-io.typeform.com/to/IVsRHfBy?typeform-medium=embed-snippet#symbol=${asset.symbol}&asset_id=${asset.id}`}
+      rel="noopener noreferrer"
+      className="parent-hover"
+      style={{
+        ['--parent-content-color' as string]: 'var(--neutral-400)',
+        ['--parent-hovered-content-color' as string]: 'var(--neutral-700)',
+      }}
+    >
+      <HStack
+        gap={8}
+        alignItems="center"
+        className="content-hover"
+        justifyContent="center"
+      >
+        <FlagIcon style={{ width: 20, height: 20 }} />
+        <UIText kind="small/accent">Report Asset</UIText>
+      </HStack>
+    </UnstyledAnchor>
+  );
+}
+
+export function AssetInfo() {
+  const { asset_code } = useParams();
+  const navigationType = useNavigationType();
+  useEffect(() => {
+    if (navigationType === NavigationType.Push) {
+      window.scrollTo(0, 0);
+    }
+  }, [navigationType]);
+  useBackgroundKind({ kind: 'white' });
```

Comment:
```suggestion
  useBackgroundKind(whiteBackground);
```

`whiteBackgroundKind` is an existing export alongside `useBackgroundKind`

Reply:
Oh, nice! Forgot about this one